### PR TITLE
Improve push constant documentation, including internal docs.

### PR DESCRIPTION
**Pull request by jimblandy**

### CODE REVIEW

File: wgpu-core/src/command/bundle.rs [hunk 1]
```diff
@@ -6,6 +6,17 @@ times, on different encoders. Constructing a render bundle lets `wgpu` validate
...
 efficient than simply re-recording its commands each time.
 
+Not all commands are available in bundles; for example, a render bundle may not
+contain a [`RenderCommand::SetViewport`] command.
+
+Most of `wgpu`'s backend graphics APIs have something like bundles. For example,
+Vulkan calls them "secondary command buffers", and Metal calls them "indirect
+command buffers". However, `wgpu`'s implementation of render bundles does not
+take advantage of those underlying platform features. At the hal level, `wgpu`
+render bundles just replay the commands.
```
Comment (cwfitzgerald) [inline wgpu-core/src/command/bundle.rs:16]: I would note that we will implement this at some point

File: wgpu-core/src/command/compute.rs [hunk 2]
```diff
@@ -43,11 +43,24 @@ pub enum ComputeCommand {
         bind_group_id: id::BindGroupId,
     },
     SetPipeline(id::ComputePipelineId),
+
+    /// Set a range of push constants to values stored in [`BasePass::push_constant_data`].
     SetPushConstant {
+        /// The byte offset within the push constant storage to write to.  This
```
Comment (cwfitzgerald) [inline wgpu-core/src/command/compute.rs:49]:
```suggestion
        /// The byte offset within the push constant storage to write to. This
```

File: wgpu/src/lib.rs [hunk 3]
```diff
@@ -3151,13 +3161,14 @@ impl<'a> ComputePass<'a> {
 
 /// [`Features::PUSH_CONSTANTS`] must be enabled on the device in order to call these functions.
 impl<'a> ComputePass<'a> {
-    /// Set push constant data.
+    /// Set push constant data for subsequent dispatch calls.
```
Comment (cwfitzgerald) [inline wgpu/src/lib.rs:3164]: These docs should reference the other docs

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Good with nits

File: wgpu/src/lib.rs (hunk 3 above)
Comment (jimblandy) [reply wgpu/src/lib.rs:3164]: Discussed in chat - the constraints on push constants in render passes don't apply to compute passes, so these docs are okay.

**Review by cwfitzgerald — APPROVED** [review]
Party on

**Pull request merged**